cognitedata/reveal
# fix: support CAD models with coordinates far from the origin

#### Type of change

![Bug](https://img.shields.io/badge/Type-Bug-red) 

## Description :pencil:


Add support for CAD models far away from origin. Most of the work in this PR is to ensure that node transforms still work as expected.

## Test instructions :information_source:



Compare this model with fusion

- Local: https://localhost:3000/?env=greenfield-3d&project=3d-test&modelId=5138454831278089&revisionId=676481897810019
- Fusion: https://cog-3d.fusion.cognite.com/3d-test/3d-models/5138454831278089/revisions/676481897810019?cluster=greenfield.cognitedata.com&env=greenfield

Currently the model "wobbles" a lot in production due to floating point precision. This PR running the local version should fix this.

As you will see when reviewing, most of the changes here are related to making sure node transform still works. I have added a transform tool in examples which can be used to click and manipulate transform of nodes (note that this requires a remote model, local will not work).
![image](https://github.com/cognitedata/reveal/assets/6595726/00dd1b8c-6edf-4921-b479-7ecb99996270)



## Checklist :ballot_box_with_check:
